geoips/plugins/modules/readers/seviri_hrit.py

```diff
@@ -194,19 +195,21 @@ def compare_dicts(d1, d2, skip=None):
 
 def get_top_level_metadata(fnames, sect):
     """Get top level metadata."""
+    md = {}
+    source_file = fnames[0]
     for fname in fnames:
         df = HritFile(fname)
         if "block_2" in df.metadata.keys():
+            source_file = fname
             break
-    md = {}
-    if "GEOS" in df.metadata["block_2"]["projection"]:
+    if "GEOS" in df.metadata.get("block_2", {}).get("projection", {}):
```

Comment:
What's the benefit of using the gets over just bracket notation?

Reply:
It returns an empty dictionary. So, if either "block_2" or "projection" doesn't exist, the test still successfully says "'GEOS' is not in the dictionary, so I will continue without processing this block".

Reply:
No, using ``dict.get(<some_key>, {})`` will result in an empty dictionary if ``<some_key>`` doesn't match a key. If you run ``dict.get(<some_key>)`` without the additional ``{}`` argument, then it will result in None and calling ``None.get()`` will raise an error.